src/Microsoft.SqlTools.ManagedBatchParser/ReliableConnection/ReliableConnectionHelper.cs

```diff
@@ -795,9 +795,24 @@ public static ServerSystemInfo GetServerCpuAndMemoryInfo(IDbConnection connectio
             {
                 SqlConnection sqlConnection = GetAsSqlConnection(connection);
                 var server = new Server(new ServerConnection(sqlConnection));
-                server.SetDefaultInitFields(server.GetType(), new String[] { nameof(server.Processors), nameof(server.PhysicalMemory) });
-                sysInfo.CpuCount = server.Processors;
-                sysInfo.PhysicalMemoryInMB = server.PhysicalMemory;
+                var defaultFields = new List<string>();
+                var isProcessorsSupported = server.IsSupportedProperty(nameof(server.Processors));
```

Comment:
is `isSupportedProperty` method a client side only operation?

Reply:
Yes, it just checks whether the property is marked as supported in the definition XML's used to create the class objects

https://github.com/microsoft/sqlmanagementobjects/blob/main/src/Microsoft/SqlServer/Management/Smo/SqlSmoObject.cs#L4385